tests/e2e/test_identity.py

```diff
@@ -0,0 +1,234 @@
+import pytest
+import json
+import logging
+import time
+
+from vdrtools import wallet, did
+from vdrtools import cheqd_keys, cheqd_pool, cheqd_ledger
+from vdrtools.error import CommonInvalidStructure
+
+from helpers import create_did_helper, query_did_helper, random_string, update_did_helper, wallet_helper, get_base_account_number_and_sequence, get_timeout_height, get_balance_vdr, send_tx_helper, \
+    SENDER_ADDRESS, SENDER_MNEMONIC, RECEIVER_ADDRESS, LOCAL_NET_NETWORK, TEST_NET_GAS_X_GAS_PRICES_INT, GAS_AMOUNT, GAS_PRICE, DENOM
+
+# logger = logging.getLogger(__name__)
+# logging.basicConfig(level=logging.DEBUG)
+
+key_alias = "operator0"
+default_amount = 1000
+default_memo = "test_memo"
+MAX_GAS_MAGIC_NUMBER_NEGATIVE = 1.2
+MAX_GAS_MAGIC_NUMBER = 1.3
+LOCAL_POOL_HTTP = "http://localhost:26657"
+FQ_PREFIX = "did:cheqd:testnet:"
+
+
+@pytest.mark.parametrize(
+    "magic_number_negative, magic_number_positive",
+    [
+        (1.2, 1.3), # base case, corner values
+        (1.0, 1.5),
+        (0.5, 2),
+    ]
+)
+@pytest.mark.parametrize("transfer_amount", ["1", "20002", "3000003", "9000000009"])
+@pytest.mark.asyncio
+async def test_gas_estimation(magic_number_negative, magic_number_positive, transfer_amount):
+    pool_alias = random_string(5)
+    await cheqd_pool.add(pool_alias, LOCAL_POOL_HTTP, LOCAL_NET_NETWORK)
+    wallet_handle, _, _ = await wallet_helper()
+    public_key = json.loads(
+        await cheqd_keys.add_from_mnemonic(wallet_handle, key_alias, SENDER_MNEMONIC, "")
+    )["pub_key"]
+
+    msg = await cheqd_ledger.bank.build_msg_send(
+        SENDER_ADDRESS, RECEIVER_ADDRESS, transfer_amount, DENOM
+    )
+    account_number, sequence_number = await get_base_account_number_and_sequence(pool_alias, SENDER_ADDRESS)
+    timeout_height = await get_timeout_height(pool_alias)
+    test_tx = await cheqd_ledger.auth.build_tx(
+        pool_alias, public_key, msg, account_number, sequence_number, 0, 0, DENOM, SENDER_ADDRESS, timeout_height, default_memo
+    )
+    request = await cheqd_ledger.tx.build_query_simulate(test_tx)
+    response = await cheqd_pool.abci_query(pool_alias, request)
+    response = await cheqd_ledger.tx.parse_query_simulate_resp(response)
+    gas_estimation = json.loads(response)["gas_info"]["gas_used"]
+
+    # negative case
```

Comment:
So, maybe it should be divided into different tests? like `test_some_positive` and `test_some_negative`.
Cause mixing several cases in the one test can be not clear.